<a href="https://colab.research.google.com/github/IfrazQazi/Email-Campaign-Effectiveness-Prediction-Capstone-Project/blob/main/Email_Campaign_Effectiveness_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Email Campaign Effectiveness Prediction**

#**Problem Statement**
##Most of the small to medium business owners are making effective use of Gmail-based Email marketing Strategies for offline targeting of converting their prospective customers into leads so that they stay with them in Business. The main objective is to create a machine learning model to characterize the mail and track the mail that is ignored; read; acknowledged by the reader.

In [1]:
#importing library
import pandas as pd 
import numpy as np
import matplotlib.pyplot  as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

%matplotlib inline 

In [2]:
import warnings
warnings.filterwarnings("ignore")

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
email_df=pd.read_csv('/content/drive/MyDrive/AlmaBetter/Email campaign Effectiveness/data_email_campaign.csv')

In [5]:
# head of dataframe
email_df.head()

Email_ID  Email_Type  ...  Total_Images  Email_Status
0  EMA00081000034500           1  ...           0.0             0
1  EMA00081000045360           2  ...           0.0             0
2  EMA00081000066290           2  ...           0.0             1
3  EMA00081000076560           1  ...           0.0             0
4  EMA00081000109720           1  ...           0.0             0

[5 rows x 12 columns]

In [6]:
email_df.tail()

Email_ID  Email_Type  ...  Total_Images  Email_Status
68348  EMA00089995974500           2  ...          13.0             0
68349  EMA00089998225300           1  ...          16.0             0
68350  EMA00089998436500           1  ...           0.0             0
68351  EMA00089999168800           1  ...           4.0             0
68352  EMA00089999316900           1  ...           0.0             0

[5 rows x 12 columns]

#**The data has the following features:**
##**Email_ID** - This column contains the email ids of individuals.

##**Email_type** - Email type contains 2 categories 1 and 2. We can assume that the types are like promotional email or sales email.

##**Subject_Hotness_Score** - It is the subject-line effectiveness score.

##**Email_Source** - It represents the source of the email like sales,marketing or product type email.

##**Email_Campaign_Type** - Campaign type

##**Total_Past_Communications** - This column contains the previous mails from the source.

##**Customer_Location** - Categorical data which explains the different demographic location of the customers.

##**Time_Email_sent_Category** - It has 3 categories: 1,2 and 3 which are considered as morning,evening and night time slot.

##**Word_Count** - It contains the no.of words in the mail.

##**Total_Links** - Total links in the email body.

##**Total_Images** - The banner images in the email body.

##**Email_Status** - It is the target variable which contains the characterization of the mail that is ignored; read; acknowledged by the reader.

In [20]:
email_df.shape

(68353, 12)

## Here we can see that this data set is consist of 12 columns/features and 68353 rows/observation

In [10]:
email_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 68353 entries, 0 to 68352
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Email_ID                   68353 non-null  object 
 1   Email_Type                 68353 non-null  int64  
 2   Subject_Hotness_Score      68353 non-null  float64
 3   Email_Source_Type          68353 non-null  int64  
 4   Customer_Location          56758 non-null  object 
 5   Email_Campaign_Type        68353 non-null  int64  
 6   Total_Past_Communications  61528 non-null  float64
 7   Time_Email_sent_Category   68353 non-null  int64  
 8   Word_Count                 68353 non-null  int64  
 9   Total_Links                66152 non-null  float64
 10  Total_Images               66676 non-null  float64
 11  Email_Status               68353 non-null  int64  
dtypes: float64(4), int64(6), object(2)
memory usage: 6.3+ MB


In [18]:
# lets see Email_ID colum
email_df['Email_ID'].count()

68353

##here we can see that Email_ID feature is having 68353 unique values and number of observation is also 68353 so we can drop that feature from our data set